pkg/chartutil/create.go

```diff
@@ -324,6 +330,17 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          {{- with .Values.env }}
```

Comment:
Why not using envFrom directly?

Reply:
here you go. i rebased the PR and put another commit on top. 

https://github.com/helm/helm/pull/11349/commits/c78d3cc42ab7673c95e1c3319bbc2aeaab622cb8

if all is fine i will squash the changes to one commit